### PR TITLE
base-images: Use new debian-base:v2.1.0 and build new debian-iptables:v12.1.0

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**
/kind cleanup
/area dependency security

**What this PR does / why we need it**:
- Fixup GCB configs for debian-{base,iptables} images
- Point `debian-base` image references to K8s Infra
- Update dependents to use `debian-base:v2.1.0`
- Build `debian-iptables:v12.1.0` image

**Which issue(s) this PR fixes**:

Fixes #
Tracking issue: https://github.com/kubernetes/kubernetes/issues/58012

**Special notes for your reviewer**:

Continuation of https://github.com/kubernetes/kubernetes/pull/90665.

**Does this PR introduce a user-facing change?**:
```release-note
- base-images: Use debian-base:v2.1.0
```

### CODE REVIEW

**Comment by justaugustus** [comment]
/assign @dims @BenTheElder 
cc: @kubernetes/release-engineering 
/priority important-soon

I have a strong feeling a bunch of these will fail because I haven't tracked down all of the `pause` image references, but I'm getting this up in the meantime to get some signal.

**Comment by justaugustus** [comment]
I rebased the `pause` changes out to focus on the debian base images.

As @BenTheElder mentioned [here](https://github.com/kubernetes/kubernetes/pull/90665#discussion_r418765117):
> what changed in pause? bumping pause is a headache

Will push a separate PR to update pause so we don't block the debian builds here.

**Comment by justaugustus** [comment]
(`pause` bumps moved to https://github.com/kubernetes/kubernetes/pull/90700.)

**Comment by justaugustus** [comment]
`pull-kubernetes-cross` is the remaining test failure, which should be fixed once the following PR merges: https://github.com/kubernetes/k8s.io/pull/835

File: build/common.sh
Comment (aojea) [inline build/common.sh:98]: this debian_iptables_version has changed in the rest of the PR, should be changed here too?

Comment (justaugustus) [reply build/common.sh:98]: @aojea -- Not yet. See here: https://github.com/kubernetes/kubernetes/pull/90697/files#diff-0ea054266fe6ea2dc38e6079e8adb98bL116

From https://github.com/kubernetes/kubernetes/pull/90665:
>     * Add base images to dependencies.yaml
>       Dependencies for each image are broken up into two sections to allow the dependency check to pass:
>       
>       * `image:tag`: versions that should be bumped in order to build the image
>       * `image:tag: dependents`: dependents that need to be bumped _after_ the image has been built and promoted

So we're bumping the tag just in the Makefile to trigger the image build for debian-iptables in this PR.
After this merges and the image is promoted, I'll come back with another PR to bring the dependents up to the newly-promotion version.

Comment (aojea) [reply build/common.sh:98]: thanks for the clarification, I'm just trying to follow this new process, seems a nice improvement

Comment (justaugustus) [reply build/common.sh:98]: Appreciate the feedback, @aojea! 💕
I'll be tracking everything here: https://github.com/kubernetes/kubernetes/issues/90698

...and documenting everything once some of the last quirks are worked out. :)

**Comment by justaugustus** [comment]
`pull-kubernetes-cross` should pass now that https://github.com/kubernetes/k8s.io/pull/835 has merged.
/test pull-kubernetes-cross

**Comment by justaugustus** [comment]
/assign @cblecker 

@BenTheElder @dims @cblecker -- This is ready for approval.
The image building fix needs to be approved as well and merged ahead of this --> kubernetes/test-infra#17496

**Comment by justaugustus** [comment]
/hold cancel
(https://github.com/kubernetes/test-infra/pull/17496 has merged. :tada:)

**Comment by neolit123** [comment]
just cleaning up some labels.

/remove-area kubeadm kubectl kubelet
/remove-sig cli cluster-lifecycle node testing
/sig release api-machinery

**Review by hasheddan — APPROVED** [review]
/lgtm

**Comment by justaugustus** [comment]
/assign @thockin 

Tim -- Do you have a little time for this debian-base image bump review? It's largely search/replace and should be quick.

File: cluster/images/etcd-empty-dir-cleanup/Dockerfile
Comment (dims) [inline cluster/images/etcd-empty-dir-cleanup/Dockerfile:15]: We need to bump version number for this image

File: cluster/images/etcd/Makefile
Comment (dims) [inline cluster/images/etcd/Makefile:70]: We need to bump version number for this image as well

**Comment by justaugustus** [comment]
> We need to bump version number for this image as well

@dims -- Thanks for the catch! I've bumped both revision numbers and opened https://github.com/kubernetes/kubernetes/issues/90772 for the etcd image bumps.

cc: @jpbetz @wenjiaswe @jingyih
/hold cancel

**Pull request merged**
